drivers/soundwire/intel.c

```diff
@@ -412,9 +414,28 @@ static int intel_shim_init(struct sdw_intel *sdw, bool clock_stop)
 	usleep_range(10, 15);
 
 	if (!clock_stop) {
+		struct sdw_bus *bus = &sdw->cdns.bus;
+		struct sdw_master_prop *prop = &bus->prop;
+		u32 syncprd;
+
+		/*
+		 * The hardware relies on an internal counter,
+		 * typically 4kHz, to generate the SoundWire SSP -
+		 * which defines a 'safe' synchronization point
+		 * between commands and audio transport and allows for
+		 * multi link synchronization. The SYNCPRD value is
+		 * only dependent on the oscillator clock provided to
+		 * the IP, so adjust based on _DSD properties reported
+		 * in DSDT tables.
+		 */
+		if (prop->mclk_freq % 6000000)
```

Comment:
yes, it looks more logical now: if the clock frequency *is* a multiple of 6MHz, use the value for 24MHg (which is probably not by a coincidense a multiple of 6MHz). I realize that this might be just a datasheet thing "do like this or it ain't work" but if there is an explanation to it here (why otherwise it's 38.4? Are these the only two possible frequencies? That isn't probably the case, in such a case you'd just use a switch.), it would be great to have it in the comment...

Reply:
I added a comment, and replaced the modulo with a simpler comparison after checking that the values are indeed 24MHz or 38.4MHz (somehow I thought in the first case it could be 6 or 12, but that's the bus clock, not the ip clock).

Thanks for the review @lyakh

Reply:
@plbossart great, thanks for the update!